lib/oauth.js

```diff
@@ -57,6 +58,21 @@ OAuth.prototype.buildParams = function (_oauth, uri, method, query, form, qsLib)
   return oa
 }
 
+OAuth.prototype.buildBodyHash = function(_oauth, body) {
+  var acceptedSignatureMethods = ['HMAC-SHA1', 'RSA-SHA1']
+  var index = acceptedSignatureMethods.indexOf(_oauth.signature_method)
+
+  if (!_oauth.signature_method || index > -1) {
```

Comment:
You can flip this condition and make it throw here. The rest of the code should be placed after that `if` statement. I think it will be easier to follow. Also you probably don't need the `index` as a separate variable, if you don't use it anywhere else.